server/user-group/user-group.model.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const find = require('lodash/find');
+const { Model } = require('sequelize');
+const { restoreOrCreate } = require('../common/database/restore');
+
+class UserGroup extends Model {
+  static fields({ DATE, INTEGER, STRING }) {
+    return {
+      parentId: {
+        type: INTEGER,
+        field: 'parent_id'
+      },
+      name: {
+        type: STRING
+      },
+      createdAt: {
+        type: DATE,
+        field: 'created_at'
+      },
+      updatedAt: {
+        type: DATE,
+        field: 'updated_at'
+      },
+      deletedAt: {
+        type: DATE,
+        field: 'deleted_at'
+      }
+    };
+  }
+
+  static associate({ EnrollmentOffering, User, UserGroupMember, OfferingUserGroup }) {
+    this.belongsToMany(User, {
+      as: 'members',
+      through: UserGroupMember,
+      foreignKey: { name: 'userGroupId', field: 'user_group_id' }
+    });
+    this.belongsToMany(EnrollmentOffering, {
+      through: OfferingUserGroup,
+      foreignKey: { name: 'enrollmentOfferingId', field: 'enrollment_offering_id' }
+    });
+    this.belongsTo(this, {
+      as: 'parent',
+      foreignKey: { name: 'parentId', field: 'parent_id' }
+    });
+    this.hasMany(this, {
+      as: 'children',
+      foreignKey: { name: 'parentId', field: 'parent_id' }
+    });
+  }
+
+  static options() {
+    return {
+      modelName: 'userGroup',
+      tableName: 'user_group',
+      timestamps: true,
+      paranoid: true,
+      freezeTableName: true
+    };
+  }
+
+  static hooks() {
+    return {
+      afterDestroy(userGroup) {
```

Comment:
Move rather to service since this will need to do a cleanup on Enrollments and other things as well down the line.